### PR TITLE
Dfg: resolve multi-driven signal ranges

### PR DESCRIPTION
**Pull request by gezalore**
Follow up from #3740, this adds a warning and resolution of multi-driven bit ranges in DFG.

Questions:
- Are you happy reusing the MULTIDRIVEN warning?
- Is the resolution OK? (see below)

The 'first' driver will be used, and later ones ignored, where first is either: the driver that drives from a lower LSB, or the driver with a smaller source location if the LSBs equal.

Notably:

assign a[8:1] = ..
assign a[9:0] = ... // This applies over the above as 0 < 1

As this is an error I think any resolution is OK. The reason we resolve this rather than leave it be is twofold:
- We want well-formed input to the rest of the passes
- We run DFG twice (pre and post inline), so we would get the MULTIDRIVEN warnings twice without resolution. I don't want to disable them on the first encounter though as we might discover something new post inlining.

### CODE REVIEW

File: test_regress/t/t_dfg_multidriver_dfg.pl [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# Copyright 2022 by Geza Lore. This program is free software; you
+# can redistribute it and/or modify it under the terms of either the GNU
+# Lesser General Public License Version 3 or the Perl Artistic License
+# Version 2.0.
+# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+
+scenarios(vlt => 1);
+
+compile(
+    fails => 1,
```
Comment (wsnyder) [inline test_regress/t/t_dfg_multidriver_dfg.pl:14]: Rename test to add _bad please.

File: src/V3DfgAstToDfg.cpp [hunk 2]
```diff
@@ -79,6 +79,18 @@ class AstToDfgVisitor final : public VNVisitor {
     // AstNode::user1p   // DfgVertex for this AstNode
     const VNUser1InUse m_user1InUse;
 
+    // TYPES
+    // Represents a driver during canonicalization
+    struct Driver {
+        FileLine* flp;
+        uint32_t lsb;
+        DfgVertex* vtxp;
```
Comment (wsnyder) [inline src/V3DfgAstToDfg.cpp:87]:
```suggestion
        FileLine* m_fileline;
        DfgVertex* m_vtxp;
        uint32_t m_lsb;
```

And comment.  The reordering is to save 4 bytes.  (I realize you just moved code, I missed this earlier)

**Review by wsnyder — COMMENTED** [review]
I'm ok with these being MULTIDRIVEN, of course that can be turned off but you do seem to test that.

File: src/V3DfgAstToDfg.cpp (hunk 2 above)
Comment (gezalore) [reply src/V3DfgAstToDfg.cpp:87]: FWIW will do this, but it does not save anything as the size of the struct needs to be a multiple of its alignment and its alignment is the largest of its members.

**Pull request merged**